browser-fetching/index.md

```diff
@@ -0,0 +1,80 @@
+---
+layout: layouts/base.njk
+title: How to fetch data using a web browser
+eleventyNavigation:
+  key: Fetch
+  order: 4
+---
+
+# Aren't you fetching?!
+
+_I'm uncomfortable, you're probably uncomfortable, let's move on to fetching in the web browser_.
+
+All modern web browsers supply a Fetch API for making a HTTP requests to a web server, and handling the data returned from the server. The Fetch API provides a programming interface to asynchronously exchange data with a web server without needing to load another web page.
+
+It's useful to web developers in scenarios such as:
+
+* **Displaying dynamic data** that changes over time
+  _For example, fetching current weather conditions for the given location_
+* **Partially updating the page** in response to user interactivity
+  _For example, fetching and displaying the current stock price for a user-provided stock index_
+* **Submitting forms** and displaying a success or error messages without needing to reload an entire page.
+  _For example, Adding an item to a shopping cart_
+
+Fetch's interface only requires a single argument—a `url` address to send the request to. The second argument is an object for specifying various options.
+
+```javascript
+  fetch(url, options) // return a Promise
+```
+As with most asynchronous APIs in JavaScript, `fetch` returns a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises) that either resolves (succeeds) or rejects (fails) after receiving a response.
+
+Below is a minimal example of using fetch to request a random dog image from [dog.ceo](https://dog.ceo). The image returned is then displayed in the page along with the full JSON response. 
+
+If you'd like to see the HTTP requests sent during fetches in your browser, open the Web Developer Tools and select the Network tab.
+
+
+```javascript
+const fetchDog = () => fetch('https://dog.ceo/api/breeds/image/random')
+  // Resolve the JSON returned from the url
+  .then(response => response.json())
+  // Display the image and the raw JSON in the page
+  .then(result => {
+    console.log(result)
+    document.querySelector('#dog').src = result.message
+    document.querySelector('#dog-results').innerHTML = JSON.stringify(result, null, 2)
+  })
+  // Log any errors to the developer console
+  .catch(error => console.error(error))
+```
+
+<script>
+const fetchDog = () => fetch('https://dog.ceo/api/breeds/image/random')
+  .then(response => response.json())
```

Comment:
This is something I'm not sure about.

Should we include inline comments, like

```javascript

const iFetchedADuck = {name: "I am a duck"} // some comment that explains this; more verbose than your usual

```

Reply:
Here is a thought. So, what would be most useful to our users eventually is to be able to use real code comments to make sense of code. If we think of this as an iterative process ,and don't even tell them, and use comments like real comments, then we get the added benefit of teaching them something "under the hood" -- something they might not realize for awhile. But remember, our non-profit is playing the long game.

Reply:
Indeed, comments are great for explanations. I've updated the example `fetchDog` function to explain a bit more what each line does. Thanks.